lib/ansible/plugins/vars/host_group_vars.py

```diff
@@ -55,60 +55,92 @@
 
 import os
 from ansible.errors import AnsibleParserError
-from ansible.module_utils.common.text.converters import to_bytes, to_native, to_text
+from ansible.module_utils.common.text.converters import to_bytes, to_native
 from ansible.plugins.vars import BaseVarsPlugin
-from ansible.inventory.host import Host
-from ansible.inventory.group import Group
+from ansible.utils.path import basedir
 from ansible.utils.vars import combine_vars
 
+CANONICAL_PATHS = {}
 FOUND = {}  # type: dict[str, list[str]]
+NAK = set()
+HOSTS = set()
+GROUPS = set()
```

Comment:
Wouldn't that make the plugin not stateless? This is why I wanted to bikeshed the attr name, because I think vars plugins are going to opt in to the feature but keep using instance attributes, which will now persist throughout a play.